spec/workers/metrics/record_db_and_es_record_counts_worker_spec.rb

```diff
@@ -0,0 +1,69 @@
+require "rails_helper"
+
+RSpec.describe Metrics::RecordDbAndEsRecordCountsWorker, type: :worker, elasticsearch: true do
+  include_examples "#enqueues_on_correct_queue", "low_priority", [0]
+
+  describe "#perform" do
+    let(:worker) { subject }
+
+    it "Records a matching count" do
+      allow(DatadogStatsClient).to receive(:increment)
+      create(:tag, :search_indexed)
+
+      Search::Tag.refresh_index
+      worker.perform
+
+      tags = { search_class: Search::Tag, db_count: 1, index_count: 1, record_difference: 0, margin_of_error: 0, action: "record_count", record_count: "match" }
+
+      expect(DatadogStatsClient).to have_received(:increment).with(
+        "elasticsearch",
+        tags: hash_including(tags),
+      )
+    end
+
+    it "Records a mismatching count and raises an error" do
+      allow(DatadogStatsClient).to receive(:increment)
+      tag = create(:tag, :search_indexed)
+
+      Search::Tag.refresh_index
+      tag.delete
+
+      expect { worker.perform }.to raise_error(Metrics::RecordDbAndEsRecordCountsWorker::DbAndEsRecordsCountMismatch)
+
+      tags = { search_class: Search::Tag, db_count: 0, index_count: 1, record_difference: 1, margin_of_error: 0, action: "record_count", record_count: "mismatch" }
+
+      expect(DatadogStatsClient).to have_received(:increment).with(
+        "elasticsearch",
+        tags: hash_including(tags),
+      )
+    end
+
+    it "Uses the db_count method on the Search class if no model is found" do
+      fake_search_class = class_double("Search::Base").as_stubbed_const(transfer_nested_constants: true)
+
+      # Piggyback off of Tags index for testing
+      stub_const "Search::Base::INDEX_ALIAS", Search::Tag::INDEX_ALIAS
+      create(:tag, :search_indexed)
+      Search::Tag.refresh_index
```

Comment:
I'm open to changing this. I'm just trying to mimic the `tags` index so that there's a valid ES index to get a count from.